app/models/ext_management_system.rb

```diff
@@ -482,15 +484,29 @@ def vm_count_by_state(state)
     vms.inject(0) { |t, vm| vm.power_state == state ? t + 1 : t }
   end
 
-  def total_vms_on;        vm_count_by_state("on");        end
+  def host_count_by_metric(metric)
+    total_metric = hosts.inject(0) { |t, host| t + host.send(metric) }
```

Comment:
I think this will end up in an n+1 because it need to access `hardware` off each host. Maybe change to `hosts.includes(:hardware)` or you may be able to add the in the `virtual_column` definition with `:uses => {:hosts => :hardware}`

@kbrock what do you think? Maybe you can suggest a more efficient way of doing this.